pkg/services/ngalert/migration/channel.go

```diff
@@ -74,14 +76,38 @@ func (om *OrgMigration) createReceiver(c *legacymodels.AlertNotification) (*apim
 		return nil, err
 	}
 
-	return &apimodels.PostableGrafanaReceiver{
+	recv := &apimodels.PostableGrafanaReceiver{
 		UID:                   c.UID,
 		Name:                  c.Name,
 		Type:                  c.Type,
 		DisableResolveMessage: c.DisableResolveMessage,
 		Settings:              data,
 		SecureSettings:        secureSettings,
-	}, nil
+	}
+	err = validateReceiver(recv, om.encryptionService.GetDecryptedValue)
+	if err != nil {
+		return nil, err
+	}
+	return recv, nil
```

Comment:
Nit: this could be simplified, the receiver should be ignored if we return an error.

```suggestion
	return recv, validateReceiver(recv, om.encryptionService.GetDecryptedValue)
```

Reply:
I'll leave that one as-is I think, I find myself slightly preferring to always return nil (or default struct) with error when possible